xla/service/gpu/runtime/nccl_clique_key.cc

```diff
@@ -37,14 +37,32 @@ namespace xla::gpu {
 // NcclCliqueKey
 //===----------------------------------------------------------------------===//
 
+namespace {
+
+// Compare by the first element.
+bool CompareGroups(const std::vector<GlobalDeviceId>& lhs,
+                   const std::vector<GlobalDeviceId>& rhs) {
+  if (rhs.empty()) return false;
```

Comment:
I think lambda should be more readable, otherwise it's a bit hard to get why sorting by `[0]` works. Also can the group be empty? I think this should never happen.